### DISCUSSION TITLE
in gui_place, why is estimated congestion and routing congestion so different?

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD#4372] [General]
I would have expected estimated routing congestion and routing congestion to be somewhat similar, but they are very, very different.

This is with the MegaBoom design:

![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/2798822/3844d7c6-25d9-4cce-93f7-f0d6f1b9b301)

![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/2798822/73812f67-ef06-4370-a56c-49bd85f8852b)

**Comment by maliberty** [comment]
I would say the general pattern of the congestion is actually quite good but the magnitude seems off.   My first guess is layer derating as the estimator doesn't know about it today.  If you try with 0 derate do you get a closer answer?

**Reply by oharboe** [reply]
Silly question, what is a derater and how do I "try with 0 derate"?

**Reply by maliberty** [reply]
In global_route.tcl comment out set_global_routing_layer_adjustment.

Augusto is looking at the correlation already and has observed the same shift.

**Reply by oharboe** [reply]
I think it makes sense to wait for Augusto to complete his investigations and possible fixes....

**Reply by gudeh** [reply]
Hi @oharboe, with the last updates on OR you should see a better correlation between the congestion estimates from RUDY and the routing congestion reported by the router. RUDY has been enhanced to recognize macros and supply nets, and we've also resolved a bug related to gcells.

However, we still need to make RUDY responsive to the "set_global_routing_layer_adjustment" (also known as "derating"). For a direct comparison between the two methods of congestion calculation, you can set the derating to 0 in flow/scripts/global_place.tcl and flow/scripts/global_route.tcl. You can check it on [a branch of mine](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/compare/master...gudeh:OpenROAD-flow-scripts:test-evalRudy).

Here's an example using tinyRocket on nangate45 with 0 derating:
![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/37420320/78a46af8-4f14-4697-94d0-f96f3c0aea98)
![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/37420320/de047d94-cf9b-4dfb-910e-a8a457cd272d)
Not so bad for an estimation, I would say.
